src/main/java/eu/europa/ted/efx/sdk1/EfxTemplateTranslatorV1.java

```diff
@@ -463,6 +463,7 @@ public void exitTemplateLine(TemplateLineContext ctx) {
     final Integer outlineNumber =
         ctx.OutlineNumber() != null ? Integer.parseInt(ctx.OutlineNumber().getText().trim()) : -1;
     assert this.stack.isEmpty() : "Stack should be empty at this point.";
+    this.stack.clear(); // Variable scope boundary. Clear declared variables 
```

Comment:
This seems to work, but it is a bit confusing with the line above: the previous line checks that "stack" is empty, then this line clears it.

Maybe instead of overriding the clear() method, it should be another method name, "clearScope" or "clearVariables", that only clears the "variableTypes" and "parameterValues" maps.

Reply:
Yes I thought about that. Then I thought that clear() should be overridden anyway. 
At the same time the assertion in the previous line was already there and I did not want to remove it.
It is there for developers. That's why it is an assertion. If t does not pass the algorithm is messed-up. 
So, that's why I kept it like this. 
Normally the assertion should be removed if we consider that the algorithm is good enough.

Reply:
One more thing. The variable scope implementation is not what it should be. It works for our current assumptions but still there is no proper variable scoping implemented at the moment. I will need to implement it properly in the future.
There are two things that I should implement better: correct variable scope and decouple the XpathAttrbuteLocator fom the translator. The seond being more important than the first one. But at least the second one will force us to go to EFX Toolkit 2.0.0 because it will change the Script Generator interface. So I am keeping these improvements until after November 14.